VFS-724: fixed bug in case of no content-length

wrey75 opened the pull request.

garydgregory commented on commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/http4/test/Http4GetContentInfoTest.java:
```diff
@@ -56,4 +59,13 @@ public void testGetContentInfo() throws FileSystemException {
+    @Test
+    public void testGetZeroContents() throws IOException {
+        final FileSystemManager fsManager = VFS.getManager();
+        final FileObject fo = fsManager.resolveFile("http4://monpremierlivre.home.blog/");
```
I think you will want to use Mockito here instead of depending on a random third party web site -- random from Apache's POV ;-)

michael-o replied:
I'd rather see a local server here because this test requires outbound access and is brittle.

wrey75 commented:
Well, I can change the URL to "http://www.apache.org/dist/" which is returning a page without content-length but doing a mock seems lots of work and there is also a test involving the "apache.org" website in the same class.

michael-o commented:
What gives you the certainty that this test won't break next month when the target changes?

garydgregory commented:
I am not a huge fan of mocking but this is a great example of when you would really want to do so.

wrey75 commented:
I have written the unit test based on the previous one found in the same class and involving the Apache website, not a mock.   
I think I've done my best with this bug: I've made a unit test which fails with the current code, made the fix and provided a PR. I know the unit test is not very stable in the long term but couldn't I rely on some guys invested in the project to write the proper test?

garydgregory commented:
Please realize that some of us here have spent a tremendous amount of time on this project already over the years.

I feel like asking you to give back a little more in the form of a complete PR in return for free and open source software is fair. You might need to adjust your expectations...

Furthermore, learning about mocking is a benefit I would not neglect. Pointing at other tests and saying these are not so great either is not, IMO, a justification for perpetuating an antipattern.

michael-o commented:
I fully agree with @garydgregory. @wrey75 Maven Wagon HTTP. Has a lot of tests with Jetty, I am sure you can take from there with chunk encoding.

wrey75 commented:
Sorry guys. I just wanted to fix a bug I've found when I tried to use the library for the first time and didn't wanted to spend a large amount of time on it. By the way, I recognize you're right: this bug is not limited to HTTP because the size of the object can be unknown.
Then I changed my mind: added a new method and a mock (for JAR files) and kept the original test with the hardcoded website. Why? Because all the other unit tests do the same!
I've then added the Mockito library and made the unit test.
